test/Mvc.Analyzers.Test/TopLevelParameterNameAnalyzerTest.cs

```diff
@@ -199,6 +213,43 @@ private async Task<Compilation> GetCompilationForGetName()
             return compilation;
         }
 
+        [Fact]
+        public async Task SpecifiesModelType_ReturnsFalse_IfModelBinderDoesNotSpecifyType()
+        {
+            var result = await RunSpecifiesModelTypeTest();
+            Assert.False(result);
```

Comment:
Just an aside, this kind of test is pretty gross. Even if it is duplicated I'd almost always rather see these methods have 5-6 lines of code so that they can be understood and stepped through. If you make one big 'run the test' method, that will accumulate complexity and start to diverge.